nixos/modules/services/monitoring/urlwatch.nix

```
@@ -0,0 +1,89 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.urlwatch;
+  format = pkgs.formats.yaml { };
+  configurationFile = format.generate "urlwatch.yaml" cfg.configuration;
+  urlsFile = lib.pipe cfg.jobs [
+    (map (format.generate "jobfile"))
+    (map builtins.readFile)
+    (builtins.concatStringsSep "---\n")
+    (pkgs.writeText "urls.yaml")
+  ];
```

```suggestion
  urlsFile = lib.concatMapStringsSep "---\n" (x: ) ....
```
i think [concatMapStringsSep](https://noogle.dev/f/lib/concatMapStringsSep) is more what you're looking for ;p